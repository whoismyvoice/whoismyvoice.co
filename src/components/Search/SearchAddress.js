import React, { Component, } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

// Component
import SearchInput from './SearchInput';
import TextFormButton from '../Buttons/TextFormButton';

class SearchAddress extends Component {
  render() {
    const {
      address,
      isStreetAddressNeeded,
      onSubmit,
      placeholder,
      zipCode,
    } = this.props;
    const inputClasses = cx('input', {
      'address_search': isStreetAddressNeeded,
    });
    let formFields = [];
    if (isStreetAddressNeeded) {
      formFields = (
        <React.Fragment>
          <div key="zipCodeDisplay" className="locked__zip">
            ZIP: {zipCode}
          </div>
          <input
            key="zipCode"
            type="hidden"
            name="zipCode"
            value={zipCode}
          />
          <input
            key="address"
            className={inputClasses}
            type="text"
            name="address"
            defaultValue={address}
            placeholder={placeholder}
          />
        </React.Fragment>
      );
    } else {
      formFields = (
        <React.Fragment>
          <SearchInput
            key="zipCode"
            name="zipCode"
          />
        </React.Fragment>
      );
    }
    return (
      <form
        method="GET"
        onSubmit={onSubmit}
      >
        {formFields}
        <div className="line-seperator"></div>
        <TextFormButton
          text="Continue"
        />
      </form>
    );
  }
}

SearchAddress.defaultProps = {
  address: '',
  placeholder: 'Enter Your Street Address',
};

SearchAddress.propTypes = {
  address: PropTypes.string,
  onSubmit: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  zipCode: PropTypes.string,
};

export default SearchAddress;
