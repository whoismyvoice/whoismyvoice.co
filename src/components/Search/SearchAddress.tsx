import * as React from 'react';
import * as cx from 'classnames';

// Component
import SearchInput from './SearchInput';
import TextFormButton from '../Buttons/TextFormButton';

interface Props {
  address?: string;
  addressErrorMessage?: string;
  isStreetAddressNeeded: boolean;
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
  placeholder?: string;
  zipCode?: string;
}

class SearchAddress extends React.Component<Props> {
  static defaultProps = {
    address: '',
    addressErrorMessage: undefined,
    isStreetAddressNeeded: false,
    onSubmit: () => undefined,
    placeholder: 'Enter Your Street Address',
    zipCode: '',
  };

  render() {
    const { address, addressErrorMessage, onSubmit, placeholder } = this.props;
    const inputClasses = cx('input', {
      address_search: this.props.isStreetAddressNeeded,
    });
    let formFields: React.ReactFragment = <React.Fragment />;
    if (this.props.isStreetAddressNeeded) {
      const zipCode = this.props.zipCode;
      formFields = (
        <React.Fragment>
          <div key="zipCodeDisplay" className="locked__zip">
            ZIP: {zipCode}
          </div>
          <input key="zipCode" type="hidden" name="zipCode" value={zipCode} />
          <input
            key="address"
            className={inputClasses}
            type="text"
            name="address"
            defaultValue={address}
            placeholder={placeholder}
          />
        </React.Fragment>
      );
    } else {
      formFields = (
        <React.Fragment>
          <SearchInput errorMessage={addressErrorMessage} name="zipCode" />
        </React.Fragment>
      );
    }
    return (
      <form method="GET" onSubmit={onSubmit}>
        {formFields}
        <div className="line-seperator" />
        <TextFormButton text="Continue" />
      </form>
    );
  }
}

export default SearchAddress;
