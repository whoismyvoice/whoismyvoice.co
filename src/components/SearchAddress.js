import React from 'react';
import SenateActions from '../actions/SenateActions';
import cx from 'classnames';

// Component
import BaseComponent from './BaseComponent';
import TextButton from './TextButton';

class SearchAddress extends BaseComponent {
  constructor() {
    super();
    this._bind('_handleChange', '_handleFocus', '_handleBlur', '_handleEnter', '_handleClick');
    this.state = {
      address: '',
      error: false,
      placeholder: 'Enter Street Name'
    };
  }
	_handleChange(event) {
    this.setState({
      address: event.target.value,
      error: false,
      placeholder: 'Enter Street Name'
    });
  }
  _handleFocus() {
    this.setState({
      placeholder: '',
      error: false
    });
  }
  _handleBlur() {
    if (this.state.address === '') {
      this.setState({
        placeholder: 'Enter Street Name'
      });
    }
  }
  _handleEnter(e) {
    if (e.keyCode === 13) {
      if (this.state.address.length < 4) {
        this.setState({
          error: true,
          placeholder: 'Enter Street Name',
          address: ''
        });
      } else {
        this.setState({
          error: false,
          placeholder: 'Enter Street Name',
          address: ''
        });
        SenateActions.fetchSpecificMember(this.state.address, this.props.zip_code, this.props.state_full);
      }
    }
  }
  _handleClick(evt) {
    evt.preventDefault();
    evt.stopPropagation();

    if (this.state.address < 4) {
      this.setState({
        error: true,
        placeholder: 'Enter Street Name',
        address: ''
      });
    } else {
      this.setState({
        error: false,
        address: ''
      });
      SenateActions.fetchSpecificMember(this.state.address, this.props.zip_code, this.props.state_full);
    }
  }
  render() {
    const inputClasses = cx(
      ['input'],
      {'error': this.state.error ||Â this.props.error}
    );

  	return <div>
  		<input
        className={inputClasses}
  			type="text"
        value={this.state.address}
  			placeholder={this.state.placeholder}
  			onChange={this._handleChange}
  			onKeyDown={this._handleEnter}
  			onFocus={this._handleFocus}
  			onBlur={this._handleBlur}
  		/>
      <div className="line-seperator"></div>
      <TextButton
        text="Continue"
        onClick={this._handleClick}
      />
  	</div>;
  }
}

SearchAddress.propTypes = {
  error: React.PropTypes.bool,
  zip_code: React.PropTypes.string
};

export default SearchAddress;
