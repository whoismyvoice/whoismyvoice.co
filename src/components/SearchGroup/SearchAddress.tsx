import React from 'react';
import cx from 'classnames';

// Component
import { SearchInput } from './SearchInput';
import TextFormButton from '../Buttons/TextFormButton';

interface Props {
  address?: string;
  addressErrorMessage?: string;
  isStreetAddressNeeded: boolean;
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
  placeholder?: string;
  zipCode?: string;
}

export class SearchAddress extends React.Component<Props> {
  static defaultProps: Props = {
    address: '',
    addressErrorMessage: undefined,
    isStreetAddressNeeded: false,
    onSubmit: () => undefined,
    placeholder: 'Enter Your Street Address',
    zipCode: '',
  };

  render(): JSX.Element {
    const { address, addressErrorMessage, onSubmit, placeholder } = this.props;
    const inputClasses = cx('input', {
      // eslint-disable-next-line @typescript-eslint/camelcase
      address_search: this.props.isStreetAddressNeeded,
    });
    let formFields: React.ReactFragment = <React.Fragment />;
    if (this.props.isStreetAddressNeeded) {
      const zipCode = this.props.zipCode;
      formFields = (
        <React.Fragment>
          <p>Multiple Congressional Representatives in</p>
          <div key="zipCodeDisplay" className="locked__zip">
            ZIP: {zipCode}
          </div>
          <input key="zipCode" type="hidden" name="zipCode" value={zipCode} />
          <input
            key="address"
            className={inputClasses}
            type="text"
            name="address"
            defaultValue={address}
            placeholder={placeholder}
          />
        </React.Fragment>
      );
    } else {
      formFields = (
        <React.Fragment>
          <SearchInput errorMessage={addressErrorMessage} name="zipCode" />
        </React.Fragment>
      );
    }
    return (
      <form method="GET" onSubmit={onSubmit}>
        {formFields}
        <div className="line-seperator" />
        <TextFormButton text="Continue" />
      </form>
    );
  }
}
