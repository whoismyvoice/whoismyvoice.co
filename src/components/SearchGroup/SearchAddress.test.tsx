import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';
import { SearchAddress } from './SearchAddress';

jest.mock('mixpanel-browser');

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SearchAddress isStreetAddressNeeded={false} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('street address not needed', () => {
  const props = {
    isStreetAddressNeeded: false,
  };

  it('renders correctly', () => {
    const tree = renderer.create(<SearchAddress {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with error message', () => {
    const tree = renderer
      .create(<SearchAddress {...props} addressErrorMessage="foo errors" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with placeholder', () => {
    const tree = renderer
      .create(<SearchAddress {...props} placeholder="fooholder" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('street address needed', () => {
  const props = {
    isStreetAddressNeeded: true,
    zipCode: '12345',
  };

  it('renders correctly', () => {
    const tree = renderer.create(<SearchAddress {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with error message', () => {
    const tree = renderer
      .create(<SearchAddress {...props} addressErrorMessage="foo errors" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with placeholder', () => {
    const tree = renderer
      .create(<SearchAddress {...props} placeholder="fooholder" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
