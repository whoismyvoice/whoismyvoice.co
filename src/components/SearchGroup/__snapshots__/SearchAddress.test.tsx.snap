// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`street address needed renders correctly 1`] = `
<form
  method="GET"
  onSubmit={[Function]}
>
  <div
    className="locked__zip"
  >
    ZIP: 
    12345
  </div>
  <input
    name="zipCode"
    type="hidden"
    value="12345"
  />
  <input
    className="input address_search"
    defaultValue=""
    name="address"
    placeholder="Enter Your Street Address"
    type="text"
  />
  <div
    className="line-seperator"
  />
  <button
    className="text-button"
    onClick={[Function]}
    type="submit"
  >
    Continue
    <span
      className="text-button-border"
    />
  </button>
</form>
`;

exports[`street address needed renders correctly with error message 1`] = `
<form
  method="GET"
  onSubmit={[Function]}
>
  <div
    className="locked__zip"
  >
    ZIP: 
    12345
  </div>
  <input
    name="zipCode"
    type="hidden"
    value="12345"
  />
  <input
    className="input address_search"
    defaultValue=""
    name="address"
    placeholder="Enter Your Street Address"
    type="text"
  />
  <div
    className="line-seperator"
  />
  <button
    className="text-button"
    onClick={[Function]}
    type="submit"
  >
    Continue
    <span
      className="text-button-border"
    />
  </button>
</form>
`;

exports[`street address needed renders correctly with placeholder 1`] = `
<form
  method="GET"
  onSubmit={[Function]}
>
  <div
    className="locked__zip"
  >
    ZIP: 
    12345
  </div>
  <input
    name="zipCode"
    type="hidden"
    value="12345"
  />
  <input
    className="input address_search"
    defaultValue=""
    name="address"
    placeholder="fooholder"
    type="text"
  />
  <div
    className="line-seperator"
  />
  <button
    className="text-button"
    onClick={[Function]}
    type="submit"
  >
    Continue
    <span
      className="text-button-border"
    />
  </button>
</form>
`;

exports[`street address not needed renders correctly 1`] = `
<form
  method="GET"
  onSubmit={[Function]}
>
  <input
    className=""
    name="zipCode"
    onChange={[Function]}
    pattern="[0-9]*"
    placeholder="Enter Your Zip Code"
    type="text"
  />
  <div
    className="line-seperator"
  />
  <button
    className="text-button"
    onClick={[Function]}
    type="submit"
  >
    Continue
    <span
      className="text-button-border"
    />
  </button>
</form>
`;

exports[`street address not needed renders correctly with error message 1`] = `
<form
  method="GET"
  onSubmit={[Function]}
>
  <p
    className="search-input-message--error"
  >
    foo errors
  </p>
  <input
    className="error"
    name="zipCode"
    onChange={[Function]}
    pattern="[0-9]*"
    placeholder="Enter Your Zip Code"
    type="text"
  />
  <div
    className="line-seperator"
  />
  <button
    className="text-button"
    onClick={[Function]}
    type="submit"
  >
    Continue
    <span
      className="text-button-border"
    />
  </button>
</form>
`;

exports[`street address not needed renders correctly with placeholder 1`] = `
<form
  method="GET"
  onSubmit={[Function]}
>
  <input
    className=""
    name="zipCode"
    onChange={[Function]}
    pattern="[0-9]*"
    placeholder="Enter Your Zip Code"
    type="text"
  />
  <div
    className="line-seperator"
  />
  <button
    className="text-button"
    onClick={[Function]}
    type="submit"
  >
    Continue
    <span
      className="text-button-border"
    />
  </button>
</form>
`;
